tools/SetupFlow/DevHome.SetupFlow/ViewModels/AddRepoViewModel.cs

```diff
@@ -981,9 +1027,14 @@ private async Task LogUserIn(string repositoryProviderName)
         ShouldShowLoginUi = true;
         IsCancelling = false;
 
+        // Store the close button text because it will change.
+        var closeButtonText = _addRepoDialog.CloseButtonText;
         _addRepoDialog.CloseButtonText = _host.GetService<ISetupFlowStringResource>().GetLocalized(StringResourceKey.UrlCancelButtonText);
+
         await InitiateAddAccountUserExperienceAsync(_providers.GetProvider(repositoryProviderName), LoginUiContent);
 
+        _addRepoDialog.CloseButtonText = closeButtonText;

```

Comment:
>_addRepoDialog.CloseButtonText = closeButtonText; [](http://example.com/codeflow?start=8&length=49)

Why are we resetting it after line 1032?

Reply:
Added comments.  This is a hold-off of adding an account and when the account login appears in the life span of the content dialog.

The flow looks like this
1. Primary button is pressed
2. Check if the user needs to log in.
3. If so, log the user in.
4. Continue with the rest of the primary button action.

Because the login happens inside the primary click event, the ContentDialog can't react to the secondary button click.  To get around this, I change the text of the secondary button to notify users to press the "X" in the upper right corner to cancel the login prompt.

An unfortunate circumstance of this code change is the ViewModel is aware of the view and how the view changes.  I am aware of this and will work toward separating View logic from ViewModel logic.